code/game/machinery/firealarm.dm

```diff
@@ -91,6 +91,7 @@
 		return
 	var/area/our_area = get_area(src)
 	RegisterSignal(our_area, COMSIG_AREA_FIRE_CHANGED, PROC_REF(handle_fire))
+	update_overlays()
```

Comment:
do not directly call update_overlays, it doesn't do anything. it's only useful in how it's used by update_icon (the fact that it doesn't means this is unneeded)